### ISSUE TITLE
Patch releases not being returned properly

### ISSUE THREAD

**Issue by jerboaa** [AdoptOpenJDK/openjdk-api-v3#313]
OpenJDK 11 did an asynchronous 11.0.9.1+1 release, which got built at AdoptOpenJDK, but the website isn't showing it properly. It appears to be an API issue where an implicit patch release number is missing in the version comparator. See:
https://github.com/AdoptOpenJDK/openjdk-api-v3/blob/master/adoptopenjdk-api-v3-models/src/main/kotlin/net/adoptopenjdk/api/v3/models/VersionData.kt#L110

It compares by major, minor, security, pre, build, adopt_build_number, optional but should compare by major, minor, security, patch, pre, build, adopt_build_number, optional

Steps to reproduce:

```
$ curl -s 'https://api.adoptopenjdk.net/v3/assets/latest/11/hotspot?release=latest&jvm_impl=hotspot&vendor=adoptopenjdk' | grep 11\\.0\\.9\\.1
<nothing>
```

Should return release info from:
https://github.com/AdoptOpenJDK/openjdk11-binaries/releases/tag/jdk-11.0.9.1%2B1

This seems to be the reason why https://adoptopenjdk.net/index.html?variant=openjdk11&jvmVariant=hotspot doesn't show the latest upstream `11.0.9.1+1` release.

**Comment by jerboaa** [comment]
Looks like https://github.com/AdoptOpenJDK/openjdk-api-v3/pull/311 was not enough?

**Comment by jerboaa** [comment]
@johnoliver @karianna Thoughts?

**Comment by jerboaa** [comment]
Seems fixed with #315 

```
curl -s 'https://staging-api.adoptopenjdk.net/v3/assets/latest/11/hotspot?release=latest&jvm_impl=hotspot&vendor=adoptopenjdk' | grep 11\\.0\\.9\\.1
[...]
        "release_name": "jdk-11.0.9.1+1",
            "openjdk_version": "11.0.9.1+1",
                "checksum_link": "https://github.com/AdoptOpenJDK/openjdk11-binaries/releases/download/jdk-11.0.9.1%2B1/OpenJDK11U-jre_arm_linux_hotspot_11.0.9.1_1.tar.gz.sha256.txt",
                "link": "https://github.com/AdoptOpenJDK/openjdk11-binaries/releases/download/jdk-11.0.9.1%2B1/OpenJDK11U-jre_arm_linux_hotspot_11.0.9.1_1.tar.gz",
                "metadata_link": "https://github.com/AdoptOpenJDK/openjdk11-binaries/releases/download/jdk-11.0.9.1%2B1/OpenJDK11U-jre_arm_linux_hotspot_11.0.9.1_1.tar.gz.json",
                "name": "OpenJDK11U-jre_arm_linux_hotspot_11.0.9.1_1.tar.gz",
            "scm_ref": "jdk-11.0.9.1+1_adopt",
        "release_name": "jdk-11.0.9.1+1",
            "openjdk_version": "11.0.9.1+1",
```

It's now just a matter of time to reach the main API endpoint as it's in staging already.

**Comment by jerboaa** [comment]
It's working on the main API endpoint now. Thanks, all!
